lib/streamlit/watcher/local_sources_watcher.py

```diff
@@ -132,10 +165,20 @@ def _register_watcher(self, filepath, module_name):
             return
 
         try:
-            wm = WatchedModule(
-                watcher=PathWatcher(filepath, self.on_file_changed),
-                module_name=module_name,
-            )
+            if is_directory:
+                # Watch the directory instead of individual files
+                watch_dir(filepath, self.on_dir_changed, glob_pattern="**/*")
```

Comment:
as requested using the watch_dir with the usage of `glob_pattern` to search all subdirectories and types of files to walk the directory files